### ISSUE TITLE
[MultiMonitor] Start menu on wrong monitor or hidden

### ISSUE THREAD

**Issue by donquixote** [linuxmint/cinnamon#5447]
I have a laptop with 2 additional external monitors.
(Cinnamon 2.8.8. I do not plan on upgrading before Cinnamon 3.x is available on my distro.)

But I do want to report that the start menu can be displaced / hidden in a multi-monitor environment, especially after wake-up from suspend-to-ram.

Sometimes this also displaces the panel itself. But this is independent from the start menu displacement.

Clicking on the "Menu" icon in the Cinnamon panel opens the main menu / start menu in the correct position above the icon.

Hitting the menu key on the keyboard opens the start menu / main menu sometimes on a different monitor than the panel, sometimes at the top of the screen (dropping down) instead of the bottom of the screen. Or it is opened in an invisible position, possible on a non-existing monitor.

I have not yet found a reproducible pattern for this behavior.
Scenarios that sometimes trigger misplaced menus or panels is to wake up with different monitors than were connected on suspend.
But sometimes it is just a regular wake-up from suspend-to-ram, with no change in monitors connected.

**Comment by janw-oostendorp** [comment]
Can confirm this bug on `3.2.7` I have setup two monitors and it always does the following.

Press _super_; menu opens on main monitor,
press _super_ again; menu closes on mon1 and opens on mon2,
press _super_ again, menu closes.
Repeat.

Menu _does_ respect the mouse position for selecting/opening an app. It will always start on the monitor the mouse is. I'd expect menu to do the same.

**Comment by JosephMcc** [comment]
@janw-oostendorp Do I understand correctly that you are using a menu applet on each monitor?

**Comment by janw-oostendorp** [comment]
Yes, 2 moniter each with a bottom pannel and each with a menu applet in the left bottom

**Comment by JosephMcc** [comment]
Thanks for clarifying. Have to go digging. Not sure if we are doing anything to deal with that particular situation.

**Comment by janw-oostendorp** [comment]
This is quite a pet peeve of mine. Dual monitor support is quite unpolished across desktop environment.
Unity handles dual monitors quite nicely.
Mate doesn't even update pannel layout when you plugin/out a extra monitor. So when on a laptop only you have two stacked pannels... Also it doesn't do anything with the _menu_ when there are two present.
Gnome 3 of course have a quite different philosophy but doesn't do anything with dual monitors.
[albert](https://albertlauncher.github.io/docs/) can handle this but I like a menu better.

If this is possible too fix I would be so happy. I'm willing too help if possible.

**Comment by callagg2** [comment]
I can also relate to this.

Except in my case, I have a Dell laptop and a monitor.
Host: Latitude-E4310 Kernel: 5.0.0-25-generic x86_64 bits: 64 
Desktop: Cinnamon 4.2.3 Distro: Linux Mint 19.2 Tina

My external monitor is set-up as the Primary screen in "cinnamon-settings display". 
But if i detach my external monitor from my laptop, i have no start menu on my laptop - opening  cinnamon-settings display (with difficulty as i have no start menu) - I cannot make this my primary display. 
In Microsoft Windows, the primary display would default back to the laptop, as it is the only available screen.

What makes this really annoying is Redshift only works on the primary display - so if i can't make my laptop the primary display, i can't get Redshift to work.

The interim solution is to open "cinnamon-settings display" and change the primary screen to my laptop before detaching the external monitor.

**Comment by callagg2** [comment]
Yes, this was fixed. thanks
